src/main/java/frc/robot/commands/WOFSpinForSameColor.java

```diff
@@ -0,0 +1,69 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.WheelOfFortune;
+
+public class WOFSpinForSameColor extends CommandBase {
+    /**
+     * Creates a new WOFSpin.
+     */
+    private final WheelOfFortune wof;
+    private String lastColor, initialColor;
+    private int currentTimes;
+    private final int times;
+    private boolean finished;
+
+    public WOFSpinForSameColor(WheelOfFortune subsystem, int times) {
+        // Use addRequirements() here to declare subsystem dependencies.
+        wof = subsystem;
+        this.times = times;
+        addRequirements(wof);
+    }
+
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+
+    }
+
+    // Called every time the scheduler runs while the command is scheduled.
+    @Override
+    public void execute() {
+        String color = wof.readColor();
+        if (lastColor == null) {
+            initialColor = color;
+            lastColor = initialColor;
+        }
+        if (!lastColor.equals(color) && color.equals(initialColor)) {
+            currentTimes++;
+        }
+        lastColor = color;
+        if (currentTimes == times) {
+            wof.wofSpinStop();
+            finished = true;
+        } else {
+            wof.wofSpin();
+        }
```

Comment:
All of this code belongs elsewhere.
isFinished can simply return this.currentTimes >= this.times;
end should call wof.wofSpinStop
the wof.wofSpin should only get called once.